tests/foreman/endtoend/test_api_endtoend.py

```diff
@@ -821,15 +821,34 @@ def test_positive_get_links(self):
         for group, path_pairs in api_paths.items():
             api_paths[group] = list(path_pairs.values())
 
-        if bz_bug_is_open(1166875):
-            # The server returns incorrect paths.
-            api_paths['docker_manifests'].append(u'/katello/api/docker_manifests')
-            api_paths['docker_manifests'].remove(u'/katello/api/compare')
-            api_paths['docker_tags'].append(u'/katello/api/docker_tags')
-            api_paths['docker_tags'].remove(u'/katello/api/compare')
-            api_paths['errata'].append(u'/katello/api/errata')
-            api_paths['errata'].append(u'/katello/api/errata/compare')
-            api_paths['errata'].remove(u'/katello/api/compare')
+            if bz_bug_is_open(1166875):
+                # The server returns incorrect paths.
+                API_PATHS['repositories'] = (
+                    u'/katello/api/repositories',
+                    u'/katello/api/repositories',
+                    u'/katello/api/repositories/:id',
+                    u'/katello/api/repositories/:id',
+                    u'/katello/api/repositories/:id',
+                    u'/katello/api/repositories/:id/export',
+                    u'/katello/api/repositories/:id/gpg_key_content',
+                    u'/katello/api/repositories/:id/import_uploads',
+                    u'/katello/api/repositories/:id/republish',
+                    u'/katello/api/repositories/:id/sync',
+                    u'/katello/api/repositories/:id/upload_content',
+                    u'/katello/api/repositories/repository_types',
+                )
+                API_PATHS['docker_manifests'] = (
+                    u'/katello/api/compare',
+                    u'/katello/api/docker_manifests/:id'
+                )
+                API_PATHS['docker_tags'] = (
+                    u'/katello/api/compare',
```

Comment:
I don't understand , what are you doing here , the workaround would be to remove the` /katello/api/compare` from server reply and add /katello/api/<component>/compare to the group